node/router/src/inbound.rs

```diff
@@ -256,7 +257,12 @@ pub trait Inbound<N: Network>: Reading + Outbound<N> {
         // Retrieve the connected peers.
         let peers = self.router().connected_peers();
         // Filter out invalid addresses.
-        let peers = peers.into_iter().filter(|ip| self.router().is_valid_peer_ip(ip)).collect();
+        let peers = if self.router().is_dev() {
+            // In dev mode, relax the validity requirements to make running devnets more flexible.
+            peers.into_iter().filter(|ip| !is_bogon_ip(ip.ip())).collect()
```

Comment:
do we want to filter out _all_ bogon addresses here? I could see private network addresses being practical in dev mode

Reply:
Good question, for this PR I basically cherry-picked what we had in place before https://github.com/AleoHQ/snarkOS/pull/2833 (and tested that it still works).